### PR TITLE
FH-3751 - Change the cmd fhc admin-users to follow the V3 standard an…

### PR DESCRIPTION
**Pull request by camilamacedo86**
…d add unit tests

JIRA: https://issues.jboss.org/browse/FH-3751

**Changes**
- Move to V3 standard
- Add unit test
- Add alias

**Steps to Verify**
- `./bin/fhc.js admin users list`  
- `./bin/fhc.js admin users read <username>`
- `./bin/fhc.js admin users delete <username>`
- ` ./bin/fhc.js admin users create <username> <email> `
- ` ./bin/fhc.js admin users update <username> <newemail> `
- `./bin/fhc.js admin users changeroles --username=<username> --roles=<roles>`
- `./bin/fhc.js admin users changeauthpolicies --username=<username> --authpolicies=<authpolicies>`
- `./bin/fhc.js admin users import <file>`
- ` ./bin/fhc.js admin users disable <username>`
- ` ./bin/fhc.js admin users enable <username>`
- `./bin/fhc.js admin users listinstallapps --username=<username>`
- `./bin/fhc.js admin users listdevices --username=<username>`

**Local Test**
- `./bin/fhc.js admin users list`  

screen shot 2017-08-17 at 2 40 24 pm

- `./bin/fhc.js admin users list --json`

screen shot 2017-08-17 at 2 40 30 pm
  
 - `./bin/fhc.js admin users read <username>`

screen shot 2017-08-17 at 2 42 41 pm

- `./bin/fhc.js admin users delete`

```
Camilas-MacBook-Pro:fh-fhc cmacedo$ ./bin/fhc.js admin users delete conor@example.com
User deleted successfully.
Camilas-MacBook-Pro:fh-fhc cmacedo$ ./bin/fhc.js admin users delete spoole+test@redhat.com --json
{
  "fields": {
    "activated": true,
    "authpolicies": [],
    "blacklisted": false,
    "businessObject": "cluster/reseller/customer/user",
    "customerRoles": [],
    "defaultDomain": "",
    "email": "spoole+test@redhat.com",
    "enabled": true,
    "fhaaa.authorisation.enabled": true,
    "groups": [],
    "hierarchy": {
      "cluster": "sam1-core",
      "cluster/reseller": "t-ub7fDklyz-gwe3oL5DPupH",
      "cluster/reseller/customer": "44WKNggefW9N0ggTtqN2SFuF",
      "cluster/reseller/customer/user": "xkqoe77zpluhzgyry3gqfne2"
    },
    "lastLogin": "2016-09-08 18:48:52",
    "name": "Shannon",
    "parentEntity": "TenantImpl:44WKNggefW9N0ggTtqN2SFuF:FeedHenry Support~support",
    "permissions": [],
    "prefs": {
      "accountType": "enterprise",
      "studio.version": "beta",
      "viewType": "grid"
    },
    "reseller": "t-ub7fDklyz-gwe3oL5DPupH",
    "resellerRoles": [],
    "roles": [],
    "storeItemGroups": [],
    "sub": "kajkiik4rxhqnnjffciyzzrn",
    "subscriber": "kajkiik4rxhqnnjffciyzzrn",
    "sysCreated": "2016-05-10 14:49:55:017",
    "sysGroupFlags": 65567,
    "sysGroupList": "",
    "sysModified": "2016-12-01 14:19:08:983",
    "sysShardPoint": 3131105919,
    "sysVersion": 98,
    "teams": [],
    "tenant": "44WKNggefW9N0ggTtqN2SFuF",
    "username": "spoole+test@redhat.com"
  },
  "guid": "xkqoe77zpluhzgyry3gqfne2",
  "status": "ok",
  "type": "ten_User"
}
Camilas-MacBook-Pro:fh-fhc cmacedo$ 
```

- To create 
```

Camilas-MacBook-Pro:fh-fhc cmacedo$ ./bin/fhc.js admin users create newuser newuser@gmail.com
User created successfully.
Camilas-MacBook-Pro:fh-fhc cmacedo$ ./bin/fhc.js admin users create newuser newuser@gmail.com --json
fhc ERR! Error creating user: User with ID already exists
fhc Command executed with error
Camilas-MacBook-Pro:fh-fhc cmacedo$ ./bin/fhc.js admin users create newuser2 newuser2@gmail.com --json
{
  "status": "ok",
  "username": "newuser2"
}
Camilas-MacBook-Pro:fh-fhc cmacedo$ 

```
- ` ./bin/fhc.js admin users update <username> <newemail> `

screen shot 2017-08-17 at 3 15 42 pm

- ` ./bin/fhc.js admin users update <username> <newemail> --json`

screen shot 2017-08-17 at 3 16 18 pm

- `./bin/fhc.js admin users changeroles --username=<username> --roles=<roles>`

screen shot 2017-08-19 at 7 51 09 pm

- `./bin/fhc.js admin users changeauthpolicies --username=<username> --authpolicies=<authpolicies>`

screen shot 2017-08-19 at 7 53 52 pm

- `./bin/fhc.js admin users import <file>`

Before the change:
```
Camilas-MacBook-Pro:fh-fhc cmacedo$ fhc admin users import test.csv 
fhc ERR! Error: Command not found: users
fhc ERR! 
fhc ERR! System Darwin 15.6.0
fhc ERR! command "/Users/cmacedo/.nvm/versions/node/v4.4.3/bin/node" "/Users/cmacedo/.nvm/versions/node/v4.4.3/bin/fhc" "admin" "users" "import" "test.csv"
fhc Command executed with error
Camilas-MacBook-Pro:fh-fhc cmacedo$ fhc admin-users import test.csv 
fhc ERR! [ { action: {},
fhc ERR!     cacheKey: '1a1fc4cfc2cb52f0c099a0b633e4d481',
fhc ERR!     error: 'Failed to parse CSV file. Error: Wrong data record format. Only 2 columns found.',
fhc ERR!     log: [ 'Failed to parse CSV file. Error: Wrong data record format. Only 2 columns found.' ],
fhc ERR!     status: 'error' } ]
fhc Command executed with error

```
After the change:

```
Camilas-MacBook-Pro:fh-fhc cmacedo$ ./bin/fhc.js admin users import test.csv 
fhc ERR! [ { action: {},
fhc ERR!     cacheKey: '6153e2dc94ec9f1c374fddfc98b0ce73',
fhc ERR!     error: 'Failed to parse CSV file. Error: Wrong data record format. Only 2 columns found.',
fhc ERR!     log: [ 'Failed to parse CSV file. Error: Wrong data record format. Only 2 columns found.' ],
fhc ERR!     status: 'error' } ]
fhc Command executed with error
Camilas-MacBook-Pro:fh-fhc cmacedo$ 
```
- With empty file.

```
Camilas-MacBook-Pro:fh-fhc cmacedo$ ./bin/fhc.js admin users import ~/Desktop/test.csv 
[{
  "action": {},
  "cacheKey": "bae97e1933411517819893f4e6a3ca23",
  "error": "",
  "log": [
    "Going to import 0 users",
    "Finished importing users"
  ],
  "progress": 100,
  "status": "complete"
}]
Camilas-MacBook-Pro:fh-fhc cmacedo$ 
```

- ` ./bin/fhc.js admin users disable <username>`

screen shot 2017-08-20 at 10 19 29 am

- ` ./bin/fhc.js admin users enable <username>`

screen shot 2017-08-20 at 10 19 33 am

### CODE REVIEW

File: lib/cmd/fh3/admin/users/listdevices.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+/* globals i18n */
+var common = require("../../../../common");
+var fhreq = require("../../../../utils/request");
+
+//TODO: Check it still been required and working - Was not possible to found data related to it
```
Comment (fh-qube) [inline lib/cmd/fh3/admin/users/listdevices.js]: INFO Complete the task associated to this TODO comment. [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AS1135)

File: lib/cmd/fh3/admin/users/listinstallapps.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
+/* globals i18n */
+var common = require("../../../../common");
+var fhreq = require("../../../../utils/request");
+
+//TODO: Check it still been required and working - Was not possible to found data related to it
```
Comment (fh-qube) [inline lib/cmd/fh3/admin/users/listinstallapps.js]: INFO Complete the task associated to this TODO comment. [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AS1135)

**Comment by fh-qube** [comment]
SonarQube analysis reported 2 issues:
* INFO 2 info

Watch the comments in this conversation to review them.

**Comment by fh-qube** [comment]
SonarQube analysis reported 2 issues:
* INFO 2 info

Watch the comments in this conversation to review them.

**Comment by fh-qube** [comment]
SonarQube analysis reported 2 issues:
* INFO 2 info

Watch the comments in this conversation to review them.

File: lib/cmd/fh3/admin/users/create.js [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    'username' : i18n._("Username of the new user."),
+    'email' : i18n._("Email of the new user. It will be used to send the invite email."),
+    'password' : i18n._("Password for this user login into RHMAP"),
+    'name' : i18n._("Name of the new user"),
+    'roles' : i18n._("Roles permissions for the user. (E.g 'sub')"),
+    'authpolicies' : i18n._("Unique 24 character GUID of Auth Policies. Run '$fhc admin policies list' to check it."),
+    'invite' : i18n._("Boolean to define if is to  send or not an email invitation for this new user"),
+    'teams' : i18n._("Unique 24 character GUID of the teams which the new user will be a member"),
+    'groups' : i18n._("Unique 24 character GUID of the groups which the new user will be a member"),
+    'json' : i18n._("Output in json format")
```
Comment (wtrocki) [inline lib/cmd/fh3/admin/users/create.js]: inconsistency. In most of the other commands we use `--json`

Comment (wtrocki) [reply lib/cmd/fh3/admin/users/create.js]: Yeah. not sure why I have added that comment. Sorry.

Comment (camilamacedo86) [reply lib/cmd/fh3/admin/users/create.js]: @wtrocki No Worry .. really tks for your review and all help so far !!!

**Pull request merged**